uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/performance/DeviceMonitor.java

```diff
@@ -0,0 +1,219 @@
+    }
+
+    /**
+     * Get the current memory usage
+     *
+     * @return the current memory usage in KB
+     */
+    public static Long getMemoryUsage() {
+        return getStats(PerformanceProfile.MEMORY, Long.class);
+    }
+
+    /**
+     * Get the current network info
+     *
+     * @return the information regarding the network usage
+     */
+    public static NetworkUsageMonitor.TrafficInfo getNetworkTrafficInfo() {
+        return getStats(PerformanceProfile.NETWORK, NetworkUsageMonitor.TrafficInfo.class);
+    }
+
+    /**
+     * Get the statistics of a particular performance profile
+     *
+     * @param performanceProfile the performance profile
+     * @param classOfT           the class of the stats result
+     * @return the current data showing its usage.
+     */
+    public static <T> T getStats(PerformanceProfile performanceProfile, Class<T> classOfT) {
+        PerformanceProfileMonitor<?> monitor = profilers.get(performanceProfile);
+
+        if (monitor == null) {
+            throw new IllegalArgumentException("The performanceProfile does not have a monitor.");
+        }
+
+        loadProcessInfo();
+
+        return classOfT.cast(monitor.getStats(processInfo));
+    }
+
+    /**
+     * Fetch the current process information
+     *
+     * @return the current process information
+     */
+    public static ProcessInfo getProcessInfo() {
+        return processInfo;
+    }
+
+    /**
+     * Updates or adds a new profiling monitor for a particular profile
+     *
+     * @param profile  the profile to update
+     * @param profiler the profile monitor to use
+     * @param <T>      the type of the data that will be collected by the {@link PerformanceProfileMonitor}
+     */
+    public static <T> void setProfiler(PerformanceProfile profile, PerformanceProfileMonitor<T> profiler) {
+        profilers.put(profile, profiler);
+    }
+
+    /**
+     * Returns the current profile monitor tool that's being used by the profile
+     *
+     * @param profile {@link PerformanceProfile}
+     * @return the profile monitor tool
+     */
+    public static PerformanceProfileMonitor<?> getProfiler(PerformanceProfile profile) {
+        return profilers.get(profile);
+    }
+
+    /**
+     * Fetches the current process information by executing several adb shell commands.
+     */
+    private static void loadProcessInfo() {
+        processInfo = new ProcessInfo();
+
+        final int pid = android.os.Process.myPid();
+
+        String command = "top -b -n 1 -p " + pid;
+        String[] commandResult = AdbShellUtils.executeShellCommand(command).split("\n");
+
+        final String[] processInfoString = commandResult[5].trim().split("\\s+");
+        final String cpuUsage = processInfoString[8];
+        final String memoryUsage = processInfoString[9];
+
+        command = "cat /proc/meminfo";
+        commandResult = AdbShellUtils.executeShellCommand(command).split("\n");
```

Comment:
Personally not a fan of re-assigning these variables...I'd rather make them private and create separate variables